utilities/esmpy/regrid_polygon_mesh.py

```diff
@@ -0,0 +1,766 @@
+                    index.append(i+1)    # (i+1)th element is the apex of the angle
+            except ZeroDivisionError:
+                raise(ZeroDivisionError("ZeroDivisionError, abs_vec1, abs_vec2 = {}, {}".format(abs_vec1, abs_vec2)))
+
+        #remove the vertices associated with zero angle
+        idx = index[0]
+        idx2 = idx * 2
+        del node_id[idx]
+        # keep the node_id contiguous
+        for i in range(len(node_id)):
+            if i >= idx:
+                node_id[i] = node_id[i] - 1
+        del lons[idx]
+        del lats[idx]
+        del lons_lats[idx2]
+        del lons_lats[idx2]  # after the first del action, element at idx2+1 shift to idx2
+        del poly_coords[idx]
+        # Handle the clockwise ordering of polygon vertices
+        node_id.reverse()
+        lons.reverse()
+        lats.reverse()
+        lons_lats.reverse()
+        poly_coords.reverse()
+
+    #add the center of the polygon to the nodes
+    elem_id = []
+    elem_id.append(int(0))
+    elemId = np.array(elem_id)
+
+    #two parametric dimensions, and two spatial dimensions
+    mesh = ESMF.Mesh(parametric_dim=2, spatial_dim=2, coord_sys=ESMF.CoordSys.SPH_DEG)
+    
+    #number of nodes equal to the number of vortices of the polygon
+    num_node = len(node_id)
+    num_elem = 1
+    nodeId = np.array(node_id)
+
+    #find max and min of the lons and lats of the polygon
+    lons_array = np.array(lons)
+    lats_array = np.array(lats)
+    lons_max = np.max(lons_array)
+    lons_min = np.min(lons_array)
+    lats_max = np.max(lats_array)
+    lats_min = np.min(lats_array)
+
+    nodeCoord = np.array(lons_lats)
+    polygon = Polygon(poly_coords)
+    nodeOwner = np.zeros(num_node)
+
+    #build up the polygon by forming connection between successive vertices
+    #no need to connect the last vertex with the first, otherwise it will give an error
+    elem_conn = []
+    elem_coord = []
+    i = 0
+    for nid in node_id:
+        elem_conn1 = node_id[i]
+        elem_conn.append(elem_conn1)
+        i += 1
+
+    elem_coordx = np.sum(lons_array)/num_node
+    elem_coordy = np.sum(lats_array)/num_node
+    elem_coord.append(elem_coordx)
+    elem_coord.append(elem_coordy)
+
+    elemConn = np.array(elem_conn)
+    elemCoord = np.array(elem_coord)
+
+    elem_type = []
+    elem_type.append(num_node)
+
+    #for polygon with vertices greater than 4, elemType is a numerical value
+    elemType=np.array(elem_type)
+
+    mesh.add_nodes(num_node,nodeId,nodeCoord,nodeOwner)
+
+    mesh.add_elements(num_elem,elemId,elemType,elemConn, element_coords=elemCoord)
+
+    return mesh, lons_min, lons_max, lats_min, lats_max, polygon
+
+
+def get_cat_geometry(aorcfile, lons_min, lons_max, lats_min, lats_max):
+    """
+    This function takes input from aorc netcdf file and the polygon boundary to set the limit
+    for the rectangle that encompasses the polygon
+    """
+
+    ds = nc4.Dataset(aorcfile)
+
+    lons_first = ds['longitude'][0]
```

Comment:
Actually, given that `latitude` and `longitude` are read n times below in `read_sub_netcdf`, it probably makes sense to store the whole of these two arrays in a global, accessing the copy instead of going to the file repeatedly.